### PR TITLE
Server-side triggering of acknowledgements in routes

### PR DESCRIPTION
**Pull request by tiagosr**
Quite simple fix for dealing with client acknowledgement. Now users can use a callback function in client code to receive immediate feedback on a given route.

### CODE REVIEW

File: lib/socket.io/router.js [hunk 1]
```diff
@@ -52,18 +52,25 @@ Router.prototype.middleware = function () {
...
+        debug('attaching ack callback to event')
+        args.push(socket.ack(packet.id))
+      }
+      
       context.event = args[0];
       context.data = args.slice(1);
 
-      co.wrap(gen).call(context, function (err) {
-        if (err) console.error(err.stack);
-        //TODO: error
```
Comment (ariporad) [inline lib/socket.io/router.js]: It looks like here you need to handle the error, or at least deal with it. Right now you just log it and continue.

**Comment by ariporad** [comment]
Hey, I commented this, but it looks like you're not actually handling the error. Is that intentional?

File: lib/socket.io/router.js (hunk 1 above)
Comment (tiagosr) [inline lib/socket.io/router.js]: @ariporad I didn't see any specific way of dealing with errors within the library, I assumed I should not force any specific way before the library was further developed, as I don't know how you were intending to manage this. Actually, my bad for not keeping your comment there - it was meant to be transferred to the .catch() callback further down.

**Comment by ariporad** [comment]
Hey, Could you merge master into your fork? It's behind, and there are merge conflicts. If the tests pass after that, I'll merge it.

**Comment by ariporad** [comment]
Sorry, one more thing: Acording to GitHub, you haven't changed anything:
screenshot

**Comment by tiagosr** [comment]
@ariporad I was aware of it, but it looks like somebody else made similar changes to mine recently - I could see the places I made the changes, but the merge went without it being any kind of conflict.

**Comment by ariporad** [comment]
Ok, Then I'm going to close this.

**Pull request closed without merging**